modules/nextflow/src/main/groovy/nextflow/executor/PbsProExecutor.groovy

```diff
@@ -66,11 +66,16 @@ class PbsProExecutor extends PbsExecutor {
             res << "ncpus=${task.config.getCpus()}".toString()
         }
         if( task.config.memory ) {
-            // https://www.osc.edu/documentation/knowledge_base/out_of_memory_oom_or_excessive_memory_usage
             res << "mem=${task.config.getMemory().getMega()}mb".toString()
         }
         if( res ) {
-            result << '-l' << "select=1:${res.join(':')}".toString()
+            if( task.config.clusterOptions && task.config.clusterOptions.indexOf('-l') != -1 ) {
+                log.warn 'cpus and memory directives are ignored when clusterOptions contains -l option'
+                log.warn 'tip: clusterOptions = { "-l select=1:ncpus=${task.cpus}:mem=${task.memory.toMega()}mb:..." }'
```

Comment:
I wonder if something smarter could be done, e.g. parse the string and add or override the values provided via task.cpus/memory etc

Reply:
See #2971 . We tried, but the PBS resource syntax is very complicated. Would require a lot of code and would likely have bugs. So I settled on this solution instead.

Reply:
I see, in this case we need to be a bit more careful with the warning. Those can produce thousands of message. Better to put into a single warning message and use instead `.warn1`  which removes duplicates

Reply:
Done